baselines/flwr_baselines/publications/leaf/femnist/femnist/dataset/dataset.py

```diff
@@ -0,0 +1,288 @@
+    """
+    train_subsets = []
+    validation_subsets = []
+    test_subsets = []
+
+    for subset in partitioned_dataset:
+        subset_len = len(subset)
+        train_len = int(train_split * subset_len)
+        # Do this checkup for full dataset use
+        # Consider the case sample size == 5 and
+        # train_split = 0.5 test_split = 0.5
+        # if such check as below is not performed
+        # one sample will be missing
+        if validation_split == 0.0:
+            test_len = subset_len - train_len
+            val_len = 0
+        else:
+            test_len = int(test_split * subset_len)
+            val_len = subset_len - train_len - test_len
+        train_dataset, validation_dataset, test_dataset = random_split(
+            subset,
+            lengths=[train_len, val_len, test_len],
+            generator=torch.Generator().manual_seed(random_seed),
+        )
+        train_subsets.append(train_dataset)
+        validation_subsets.append(validation_dataset)
+        test_subsets.append(test_dataset)
+    return train_subsets, validation_subsets, test_subsets
+
+
+def transform_datasets_into_dataloaders(
+    datasets: List[Dataset], **dataloader_kwargs
+) -> List[DataLoader]:
+    """
+    Transform datasets into dataloaders.
+    Parameters
+    ----------
+    datasets: List[Dataset]
+        list of datasets
+    dataloader_kwargs
+        arguments to DataLoader
+
+    Returns
+    -------
+    dataloader: List[DataLoader]
+        list of dataloaders
+    """
+    dataloaders = []
+    for dataset in datasets:
+        dataloaders.append(DataLoader(dataset, **dataloader_kwargs))
+    return dataloaders
+
+
+# pylint: disable=too-many-arguments
+def create_federated_dataloaders(
+    sampling_type: str,
+    dataset_fraction: float,
+    batch_size: int,
+    train_fraction: float,
+    validation_fraction: float,
+    test_fraction: float,
+    random_seed: int,
+) -> Tuple[List[DataLoader], List[DataLoader], List[DataLoader]]:
+    """Create the federated dataloaders by following all the preprocessing
+    steps and division.
+
+    Parameters
+    ----------
+    sampling_type: str
+        "niid" or "iid"
+    dataset_fraction: float
+        fraction of the total data that will be used for sampling
+    batch_size: int
+        batch size
+    train_fraction, validation_fraction, test_fraction: float
+        fraction of each local dataset used for training, validation, testing
+    random_seed: int
+        random seed for data shuffling
+
+    Returns
+    -------
+    """
+    if train_fraction + validation_fraction + test_fraction != 1.0:
+        raise ValueError(
+            "The fraction of train, validation and test should add up to 1.0."
+        )
+    # Download and unzip the data
+    log(INFO, "NIST data downloading started")
+    nist_by_class_url = "https://s3.amazonaws.com/nist-srd/SD19/by_class.zip"
+    nist_by_writer_url = "https://s3.amazonaws.com/nist-srd/SD19/by_write.zip"
```

Comment:
I'd stick to the current implementation since the variable are used only in a single function. I don't think that the global variable are generally a good practice